src/vos/vos_iterator.c

```diff
@@ -649,6 +672,20 @@ vos_iterate(vos_iter_param_t *param, vos_iter_type_t type, bool recursive,
 			}
 		}
 
+		if (post_cb) {
+			rc = post_cb(ih, &iter_ent, type, param, arg, &acts);
+			if (rc != 0)
+				break;
+
+			set_reprobe(type, acts, anchors, param->ip_flags);
+
+			if (need_reprobe(type, anchors)) {
```

Comment:
This could be moved out from the 'if (post_cb)', so it could be changed like: after subtree iteration, post_cb, we do a need_reprobe check to see if a reprobe is required in current level.

Reply:
Not sure I follow this.

Reply:
I think the one after subtree iteration could be skipped, right? Here we've got the iterator entry, it can go invalid only when the entry is deleted by iterator callbacks, and I don't think subtree iteration can delete parent entry now.

Reply:
I think that sounds right.